homeassistant/util/dt.py

```diff
@@ -337,12 +337,23 @@ def _lower_bound(arr: list[int], cmp: int) -> int | None:
     # For example: if triggering on 2:30 and now is 28.10.2018 2:30 (in DST)
     # we should trigger next on 28.10.2018 2:30 (out of DST), but our
     # algorithm above would produce 29.10.2018 2:30 (out of DST)
+
+    # Step 1: Check if now is ambiguous
     if _datetime_ambiguous(now):
         check_result = find_next_time_expression_time(
             now + _dst_offset_diff(now), seconds, minutes, hours
         )
+
+        # Step 2: Check if result of (now - DST) is ambiguous.
         if _datetime_ambiguous(check_result):
-            return check_result
+            # Override the DST to DST-off
+            check_result = check_result.replace(fold=1)
+
+            # Step 3: Check if check_result is in the future and earlier than result
+            if as_utc(check_result) > as_utc(now) and as_utc(check_result) < as_utc(
+                result
+            ):
```

Comment:
Should we store the result from `as_utc(check_result)` so we only have to convert it once?

Reply:
I mean compared to all the other conversions for this edge case that should be no concern... Like every dt ambiguous check has to convert the time...